Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,946 @@
+        private static object entitiesLock = new object();
+        private static SortedDictionary<string, char> entities;
+        private static Lazy<HttpEncoder> defaultEncoder;
+        private static Lazy<HttpEncoder> currentEncoderLazy;
+        private static HttpEncoder currentEncoder;
+
+        private static IDictionary<string, char> Entities
+        {
+            get
+            {
+                lock (entitiesLock)
+                {
+                    if (entities == null)
+                    {
+                        InitEntities();
+                    }
+
+                    return entities;
+                }
+            }
+        }
+
+        public static HttpEncoder Current
+        {
+            get
+            {
+                if (currentEncoder == null)
+                {
+                    currentEncoder = currentEncoderLazy.Value;
+                }
+
+                return currentEncoder;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                currentEncoder = value;
+            }
+        }
+
+        public static HttpEncoder Default
+        {
+            get
+            {
+                return defaultEncoder.Value;
+            }
+        }
+
+        private static HttpEncoder()
+        {
+            defaultEncoder = new Lazy<HttpEncoder>(() => new HttpEncoder());
+            currentEncoderLazy = new Lazy<HttpEncoder>(new Func<HttpEncoder>(GetCustomEncoderFromConfig));
+        }
+
+        public HttpEncoder()
+        {
+        }
+
+        protected internal virtual void HeaderNameValueEncode(
+            string headerName,
+            string headerValue,
+            out string encodedHeaderName,
+            out string encodedHeaderValue)
+        {
+            if (String.IsNullOrEmpty(headerName))
+            {
+                encodedHeaderName = headerName;
+            }
+            else
+            {
+                encodedHeaderName = EncodeHeaderString(headerName);
+            }
+
+            if (String.IsNullOrEmpty(headerValue))
+            {
+                encodedHeaderValue = headerValue;
+            }
+            else
+            {
+                encodedHeaderValue = EncodeHeaderString(headerValue);
+            }
+        }
+
+        private static void StringBuilderAppend(string s, ref StringBuilder sb)
+        {
+            if (sb == null)
+            {
+                sb = new StringBuilder(s);
+            }
+            else
+            {
+                sb.Append(s);
+            }
+        }
+
+        private static string EncodeHeaderString(string input)
+        {
+            StringBuilder sb = null;
+
+            for (int i = 0; i < input.Length; i++)
+            {
+                char ch = input[i];
+
+                if ((ch < 32 && ch != 9) || ch == 127)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)